### PR TITLE
Allow admin to clone and push to all repositories

### LINKED ISSUE #2452: Admin user should be able to clone/push to all repositories
I think the admin user should be able to clone and push to all repositories without being in a organisation and team.

This is somewhat related to this issue: https://github.com/gogits/gogs/issues/1146

### PR DESCRIPTION
**Pull request by roblabla**
Fixes #2452. Also allows admin to see the private repos on the user's profile. 

Tested with SSH login.

### CODE REVIEW

**Comment by bkcsoft** [comment]
The general concensus in that issue seemed to be no push-rights though. Otherwise looks good :smiley:

**Comment by roblabla** [comment]
Well, that's a bit strange. The admin already has access to lots and lots of things (Like the settings panel). Why restrict push access ?

**Comment by bkcsoft** [comment]
The reason for not allowing push-rights is explained in the comments for that Issue :)

**Comment by roblabla** [comment]
I don't agree with the line of reasoning presented there. As I explained, the admin already has "too much access". To me, the administrator should be allowed to do whatever he wants, that's the whole point of that role.

I need that level of access for a bot that's running on my gogs instance (it automatically pushes changes to READMEs). Having to do a AddPrivilege->Push->RemovePrivilege is a bit of a bother. Not terrible, just annoying. Furthermore, so long as #2780 isn't merged, I can't even acquire those privileges through the API.

File: models/access.go [hunk 1]
```diff
@@ -67,6 +67,10 @@ func accessLevel(e Engine, u *User, repo *Repository) (AccessMode, error) {
 		return mode, nil
 	}
 
+	if u.IsAdmin {
+		return ACCESS_MODE_ADMIN, nil
+	}
+
 	if u.Id == repo.OwnerID {
```
Comment (Unknwon) [inline models/access.go]: Please merge the if condition.

File: models/access.go [hunk 2]
```diff
@@ -67,6 +67,10 @@ func accessLevel(e Engine, u *User, repo *Repository) (AccessMode, error) {
...
 		return ACCESS_MODE_OWNER, nil
 	}
```
Comment (Unknwon) [inline models/access.go]: Use `else if` should be better.

File: models/access.go [hunk 3]
```diff
@@ -67,7 +67,11 @@ func accessLevel(e Engine, u *User, repo *Repository) (AccessMode, error) {
 		return mode, nil
 	}
 
-	if u.Id == repo.OwnerID {
+	else if u.IsAdmin {
+		return ACCESS_MODE_ADMIN, nil
+	}
+
```
Comment (Unknwon) [inline models/access.go]: Please make sure it compiles.

**Pull request closed without merging**